### PR TITLE
initial datawarehouse query with comments

### PR DESCRIPTION
**Pull request by Malthe122**

### CODE REVIEW

File: datawarehouse/intialBuild.sql [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	wind_km_ph float,
+	wind_direction varchar(2),
+	fog_percent float,
+	sunshine_w_m float,
+	rain_mm int,
+	road_quality road_quality_type
+	-- road_type road_type
+);
+
+CREATE TABLE consumption (
```
Comment (Malthe122) [inline datawarehouse/intialBuild.sql:40]: make this into fact. Reintroduce outcommented attributes

File: datawarehouse/intialBuild.sql [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	avg_consumption_wh_km int,
+	"name" varchar(255),
+	ac_power int,
+	"power" int,
+	producer_name varchar(255), -- should this not be a foreign key to a producer?
+	"year" int,
+	pt_effeciency int
+);
+
+CREATE TABLE vehicle (
```
Comment (Malthe122) [inline datawarehouse/intialBuild.sql:22]: rename to vehicle_trip_status. Contain attributes that will be identical throughout the whole trip (like additional car load)

File: datawarehouse/intialBuild.sql [hunk 3]
```diff
@@ -0,0 +1,65 @@
+
+
+CREATE TYPE battery_size_type AS ENUM ('small', 'medium', 'large');
+CREATE TYPE road_quality_type AS ENUM ('bad', 'medium', 'good');
```
Comment (Malthe122) [inline datawarehouse/intialBuild.sql:4]: should be int range, if we can even fetch this data from anywhere

File: datawarehouse/intialBuild.sql [hunk 4]
```diff
@@ -0,0 +1,65 @@
+
+
+CREATE TYPE battery_size_type AS ENUM ('small', 'medium', 'large');
```
Comment (Malthe122) [inline datawarehouse/intialBuild.sql:3]: should be kilowatt size (float or int)

File: datawarehouse/intialBuild.sql (hunk 2 above)
Comment (Malthe122) [reply datawarehouse/intialBuild.sql]: Done

File: datawarehouse/intialBuild.sql (hunk 3 above)
Comment (Malthe122) [reply datawarehouse/intialBuild.sql]: Done

File: datawarehouse/intialBuild.sql (hunk 1 above)
Comment (Malthe122) [reply datawarehouse/intialBuild.sql]: Done

File: datawarehouse/intialBuild.sql (hunk 4 above)
Comment (Malthe122) [reply datawarehouse/intialBuild.sql]: Done

File: datawarehouse/intialBuild.sql [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	additional_weight_grams int,
+	vehicle_milage_meters int,
+	-- driver_aggresiveness int
+);
+
+CREATE TABLE weather (
+	id serial PRIMARY KEY,
+	temperature_celcius float,
+	wind_km_ph float,
+	wind_direction varchar(2),
```
Comment (jokerofun) [inline datawarehouse/intialBuild.sql:38]: Why varchar(2)? How do we represent the direction with only two characters?
I would expect strings such as "north", "south", etc.

File: datawarehouse/intialBuild.sql [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	fog_percent float,
+	sunshine_w_m float,
+	rain_mm int,
+	road_quality int, -- TODO we need to define a range or something here, if it does not exists
+	road_type road_type
+);
```
Comment (jokerofun) [inline datawarehouse/intialBuild.sql:44]: How about humidity? I believe that it has some impact on the battery life.

File: datawarehouse/intialBuild.sql [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+CREATE TABLE fact_travel(
+	speed_km_per_hour float,
+	weather_id int REFERENCES weather(id),
+	edge_id int,
+	edge_percent float,
+	timestamp_id int REFERENCES timestamp(id),
+	acceleration_metre_per_second_squared float,
+	energy_consumption_Kwh float,
```
Comment (jokerofun) [inline datawarehouse/intialBuild.sql:68]: I see that we have another fact table "fact_consumption" which has an "energy_use" column. Does it make sense to have the same column in this fact table?

File: datawarehouse/intialBuild.sql [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+CREATE TABLE producer (
+	id serial primary key,
+	name varchar(255)
+)
+
+CREATE TABLE vehicle_trip_status (
+	id serial PRIMARY KEY,
+	vehicle_model_id int REFERENCES vehicle_model(id),
+	additional_weight_grams int,
```
Comment (sobibence) [inline datawarehouse/intialBuild.sql:29]: i think it should be in kilogramms

File: datawarehouse/intialBuild.sql [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+	id serial PRIMARY KEY,
+	edge_id int,
+	day_in_year smallint,
+	minute_in_day smallint,
+	vehicle_id int REFERENCES vehicle(id),
+	weather_id int REFERENCES weather(id),
+	energy_use_wh float
+);
+
+CREATE TABLE timestamp (
```
Comment (sobibence) [inline datawarehouse/intialBuild.sql:56]: we dont need this table better to use the postgres timestamp type

File: datawarehouse/intialBuild.sql [hunk 10]
```diff
@@ -0,0 +1,71 @@
+
+
+CREATE TYPE road_type AS ENUM ('asphalt'); -- lets add these as we are introduced to them by the road data?
+
+CREATE TABLE vehicle_model (
+	id serial PRIMARY KEY,
+	battery_size_kwh int,
+	rolling_resistance int,
+	drag_coefficient int,
+	frontal_size int,
```
Comment (sobibence) [inline datawarehouse/intialBuild.sql:10]: float

File: datawarehouse/intialBuild.sql [hunk 11]
```diff
@@ -0,0 +1,71 @@
+
+
+CREATE TYPE road_type AS ENUM ('asphalt'); -- lets add these as we are introduced to them by the road data?
+
+CREATE TABLE vehicle_model (
+	id serial PRIMARY KEY,
+	battery_size_kwh int,
+	rolling_resistance int,
+	drag_coefficient int,
```
Comment (sobibence) [inline datawarehouse/intialBuild.sql:9]: float

File: datawarehouse/intialBuild.sql [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+	frontal_size int,
+	weight_gram int,
```
Comment (sobibence) [inline datawarehouse/intialBuild.sql:11]: also in kg

**Review by sobibence — CHANGES_REQUESTED** [review]

File: datawarehouse/intialBuild.sql (hunk 7 above)
Comment (Malthe122) [reply datawarehouse/intialBuild.sql:68]: I believed we discussed this at project meeting yesterday. Will leave it unresolved, so you can determine it

File: datawarehouse/intialBuild.sql (hunk 11 above)
Comment (Malthe122) [reply datawarehouse/intialBuild.sql:9]: you marked 3 lines of code, but you mean the drag_coefficient right?

File: datawarehouse/intialBuild.sql (hunk 10 above)
Comment (Malthe122) [reply datawarehouse/intialBuild.sql:10]: made frontal_size a float

File: datawarehouse/intialBuild.sql [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+	fog_percent float,
+	sunshine_w_m float,
+	rain_mm int,
+	road_quality int, -- TODO we need to define a range or something here, if it does not exists
+	road_type road_type
+);
+
+CREATE TABLE fact_consumption (
+	id serial PRIMARY KEY,
+	edge_id int,
```
Comment (sobibence) [inline datawarehouse/intialBuild.sql:46]: this should be a reference to the edge table

File: datawarehouse/intialBuild.sql [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+	minute_in_day smallint,
+	vehicle_id int REFERENCES vehicle(id),
+	weather_id int REFERENCES weather(id),
+	energy_use_wh float
+);
+
+CREATE TABLE fact_travel(
+	speed_km_per_hour float,
+	weather_id int REFERENCES weather(id),
+	edge_id int,
```
Comment (sobibence) [inline datawarehouse/intialBuild.sql:57]: this also should be a refence

**Review by sobibence — CHANGES_REQUESTED** [review]

**Review by sobibence — APPROVED** [review]

File: datawarehouse/intialBuild.sql (hunk 7 above)
Comment (Malthe122) [reply datawarehouse/intialBuild.sql:85]: i will change it to wh

Comment (Malthe122) [reply datawarehouse/intialBuild.sql:85]: rename the other one to have the same name

File: datawarehouse/intialBuild.sql [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+	road_quality int, -- TODO we need to define a range or something here, if it does not exists
+	road_type road_type
+);
+
+CREATE SCHEMA map;
+SET search_path = map;
+
+CREATE TABLE node(
+	id serial PRIMARY KEY, -- we will have this beside the key from provider to make sure we are able to store data from a different provider in the future
+	latitude float,
```
Comment (Malthe122) [inline datawarehouse/intialBuild.sql:49]: make these doubles

File: datawarehouse/intialBuild.sql (hunk 13 above)
Comment (Malthe122) [reply datawarehouse/intialBuild.sql]: Done

**Pull request merged**